src/core/connection.c

```diff
@@ -6567,6 +6660,217 @@ QuicConnParamSet(
         return QUIC_STATUS_SUCCESS;
     }
 
+    case QUIC_PARAM_CONN_ADD_LOCAL_ADDRESS: {
+
+        if (BufferLength != sizeof(QUIC_ADDR)) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        if (QuicConnIsServer(Connection)) {
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
+
+        if (!Connection->State.Started) {
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
+
+        QUIC_CID_LIST_ENTRY* NewDestCid = QuicConnGetUnusedDestCid(Connection);
+        if (NewDestCid == NULL) {
+            Status = QUIC_STATUS_INVALID_STATE;
+            break;
+        }
+
+        const QUIC_ADDR* LocalAddress = (const QUIC_ADDR*)Buffer;
+
+        if (!QuicAddrIsValid(LocalAddress)) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+
+        BOOLEAN AddrInUse = FALSE;
+        for (uint8_t i = 0; i < Connection->PathsCount; ++i) {
+            if (QuicAddrCompare(
+                &Connection->Paths[i].Route.LocalAddress,
+                LocalAddress)) {
+                AddrInUse = TRUE;
+                break;
+            }
+        }
+
+        if (AddrInUse) {
+            Status = QUIC_STATUS_ADDRESS_IN_USE;
+            break;
+        }
+
+        if (Connection->PathsCount == QUIC_MAX_PATH_COUNT) {
```

Comment:
Put this check up before creating a new destination CID. Also, return 'invalid state' instead?

Reply:
I agree with you. I will change this.